File: backend/prompts/characters.py
from models import CharacterId

BOARD_LOCATIONS: list[list[str]] = [
    ["Arrakeen", "Carthag", "Giedi Prime"],
    ["Sietch Tabr", "The Palace", "Salusa Secundus"],
    ["Jacurutu", "Tuono Basin", "Heighliner"],
]

CHARACTERS: dict[CharacterId, dict[str, str]] = {
    "baron_harkonnen": {
        "name": "Baron Vladimir Harkonnen",
        "difficulty": "hard",
        "personality": (
            "You are vicious, theatrical, and dripping with aristocratic contempt. "
            "You treat your opponent like a bug beneath your suspensor chair. "
            "Reference scheming, betrayal, and your absolute dominance. "
            "Mock their pathetic strategy. Make them feel worthless for even trying."
        ),
    },
    "reverend_mother": {
        "name": "Reverend Mother Superior",
        "difficulty": "medium",
        "personality": (
            "You are cold, surgical, and terrifyingly perceptive. "
            "You demolish your opponent with Bene Gesserit precision disguised as prophecy. "
            "Reference prescience and the Voice. Make them feel like a lab specimen "
            "whose every move was predicted centuries ago. Clinical devastation."
        ),
    },
    "stilgar": {
        "name": "Stilgar",
        "difficulty": "easy",
        "personality": (
            "You are a bewildered Fremen warrior genuinely confused by how bad "
            "your opponent is. You apply desert wisdom to the game and accidentally "
            "devastate them with honest observations. Reference sandworms, water "
            "discipline, and the Fremen way. Your sincerity makes the roast worse."
        ),
    },
}

FALLBACK_COMMENTARY = "The spice... clouds my vision. I place my mark here."

File: frontend/src/data/characters.ts
import type { Character } from '../types';

/**
 * CPU opponent definitions matching the backend's three characters.
 * See game-design.md for full personality details and difficulty instructions.
 */
export const CHARACTERS: readonly Character[] = [
  {
    id: 'baron_harkonnen',
    name: 'Baron Harkonnen',
    difficulty: 'hard',
    difficultyRank: 3,
    description:
      'Cruel and theatrical. Treats the game as a display of dominance, mocking every move you make.',
  },
  {
    id: 'reverend_mother',
    name: 'Reverend Mother',
    difficulty: 'medium',
    difficultyRank: 2,
    description:
      'Cryptic and unsettling. Speaks in Bene Gesserit riddles and makes you feel your moves were predicted.',
  },
  {
    id: 'stilgar',
    name: 'Stilgar',
    difficulty: 'easy',
    difficultyRank: 1,
    description:
      'Earnest and confused. Applies Fremen desert wisdom to game strategy — poorly.',
  },
] as const;

/** Look up a character by ID. */
export function getCharacter(id: string): Character | undefined {
  return CHARACTERS.find((c) => c.id === id);
}
